synfig-studio/src/gui/states/state_bone.cpp

```diff
@@ -626,57 +623,67 @@ StateBone_Context::event_mouse_release_handler(const Smach::event& x)
-					bool is_currently_on(get_work_area()->get_type_mask()&Duck::TYPE_WIDTH);
-					if(is_currently_on){
-						get_canvas_view()->toggle_duck_mask(Duck::TYPE_WIDTH);
-					}
+				if(skel_layer || deform_layer){ // if selected layer is a Skeleton Layer or a Skeleton Deformation Layer
+					update_tool_options(skel_layer? SKELETON_TYPE : SKELETON_DEFORMATION_TYPE);
+					update_width_duck_status(skel_layer? SKELETON_TYPE : SKELETON_DEFORMATION_TYPE);
 
-					createChild->set_param("canvas",skel_layer->get_canvas());
+					createChild->set_param("canvas",layer->get_canvas());
 					ValueDesc list_desc(layer,"bones");
+
 					int b = -1;
 					if((clickOrigin-releaseOrigin).mag()<0.01)
 						b=find_bone(clickOrigin,layer);
-					if(b!=-1){ //! if bone found around the release point --> set active bone
+
+					if(b!=-1){ // if bone found around the release point, then set it as active bone
 						active_bone=b;
 						ValueNode_StaticList::Handle list_node;
-						list_node=ValueNode_StaticList::Handle::cast_dynamic(list_desc.get_value_node());
-						ValueDesc value_desc= ValueDesc(list_node,active_bone,list_desc);
-						setActiveBone->set_param("active_bone_node",value_desc.get_value_node());
+						list_node = ValueNode_StaticList::Handle::cast_dynamic(list_desc.get_value_node());
+						ValueDesc value_desc = ValueDesc(list_node,active_bone,list_desc);
+						ValueNode::Handle bone_value_node = value_desc.get_value_node();
+						if (deform_layer) {
+							ValueNode_Composite::Handle comp = ValueNode_Composite::Handle::cast_dynamic(value_desc.get_value_node());
+							bone_value_node = comp->get_link("first");
+						}
+						setActiveBone->set_param("active_bone_node",bone_value_node);
 						setActiveBone->set_param("prev_active_bone_node",get_work_area()->get_active_bone_value_node());
 						if(setActiveBone->is_ready()){
 							try{
 								get_canvas_interface()->get_instance()->perform_action(setActiveBone);
 							} catch (...) {
-								info("Error performing action");
+								get_canvas_interface()->get_ui_interface()->error(_("Error setting the new active bone"));
 							}
 						}
 					}
-					else{
+					else{ // create a child bone
 						ValueNode_StaticList::Handle list_node;
 						list_node=ValueNode_StaticList::Handle::cast_dynamic(list_desc.get_value_node());
 						if(active_bone!=-1 && !list_node->list.empty()){ //! if active bone is already set
 							ValueDesc value_desc= ValueDesc(list_node,active_bone,list_desc);
 							ValueNode_Bone::Handle bone_node;
+							if (deform_layer) {
+								ValueNode_Composite::Handle comp = ValueNode_Composite::Handle::cast_dynamic(value_desc.get_value_node());
+								value_desc = ValueDesc(comp,comp->get_link_index_from_name("first"),value_desc);
+							}
+
 							if (!(bone_node = ValueNode_Bone::Handle::cast_dynamic(value_desc.get_value_node())))
 							{
-								error("expected a ValueNode_Bone");
+								error("2. expected a ValueNode_Bone");
```

Comment:
Why "2. " at the start of the string? ^__^

Reply:
Oh, it wasn't supposed to be committed.

I was investigating a crash and numbering all those strings. From the crash fix, I did this partial PR. Next it'll be the actual crash fix.

Reply:
cleaned up now and rebased! :)